staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -541,6 +541,19 @@ func InClusterConfig() (*Config, error) {
 	}, nil
 }
 
+// InClusterNamespace return the default namespace to be used
```

Comment:
_if_ we expose this directly, it seems like it should
1. have similar behavior to InClusterConfig() in returning `ErrNotInCluster` if run outside a pod environment
2. have similar behavior to `inClusterClientConfig#Namespace()` in handling of `POD_NAMESPACE` and `/var/run/secrets/kubernetes.io/serviceaccount/namespace`

Reply:
I fix it, please take another look.